common/lib/common-utils/src/timer.ts

```diff
@@ -143,7 +168,15 @@ export class Timer implements ITimer {
             duration,
             intendedDuration,
             handler,
-            timeout: setTimeout(() => this.handler(), duration),
+            timeout: setLongTimeout(
+                () => this.handler(),
+                duration,
+                (timer: number) => {
+                    if(this.runningState !== undefined) {
```

Comment:
nit: space after if.
Can you add that this class supports long timeouts in the class description?